src/service.py

```diff
@@ -135,11 +136,29 @@ def cluster_sentences(self, language_name: str, save_path: str, sentences: List[
             sent_vectors.append(to_array.mean(axis=0).tolist())
 
         # third using kmeans to cluster
-        kmeans = KMeans(n_clusters=n_clusters, random_state=0).fit(sent_vectors)
-        labels = kmeans.labels_
+        best_score, best_labels = -1, None
+        evaluator = Evaluator(sent_vectors)
+        labels1 = evaluator.kmeans_strategy(n_clusters)
+        score1 = evaluator.higher_better_score(labels1)
+        labels2 = evaluator.agglomerative_strategy(n_clusters)
+        score2 = evaluator.higher_better_score(labels2)
+        if score1 < score2:
+            best_score = score2
+            best_labels = labels2
+            print('agglomerative is better than kmeans')
+        else:
+            best_score = socre1
```

Comment:
Varible name mismatch of score1

Reply:
Good find @harisalam. @jackzhenguo did you get a chance to edit this, if not I will do so.